virtcontainers/container.go

```diff
@@ -1267,6 +1267,56 @@ func (c *Container) updateResources(oldResources, newResources ContainerResource
 		// recalculate the actual number of vCPUs if a different number of vCPUs was removed
 		newResources.VCPUs = oldVCPUs - vcpusRemoved
 	}
+	return nil
+}
+
+func (c *Container) updateMemoryResources(oldResources, newResources ContainerResources) error {
+	oldMem := oldResources.Mem
+	newMem := newResources.Mem
+
+	if newMem == 0 || oldMem == newMem {
+		c.Logger().WithFields(logrus.Fields{
+			"old-mem": fmt.Sprintf("%dMB", oldMem),
+			"new-mem": fmt.Sprintf("%dMB", newMem),
+		}).Debug("the actual number of Mem will not be modified")
+		return nil
+	}
+
+	if oldMem < newMem {
+		// hot add memory
+		addMemMB := newMem - oldMem
+		// hot add memory must be aligned to 128MB
+		if addMemMB%128 != 0 {
+			return fmt.Errorf("can not hot add memory which isn't aligned to 128MB")
```

Comment:
Can you change the error message to something like `Memory must be aligned to 128 MB for hotplug` ? or `hot add` ?

Reply:
ok， Updated